### PR TITLE
Causal fidelity for tabular

### PR DESCRIPTION
**Pull request by DavidPetiteau**
# Add causal fidelity metrics for tabular data

The subject of this pull request is to add causal fidelity metrics for tabular data (deletion and insertion):
- Add metrics 318a019
- Add docs 7809cd8
- Add unit tests 98f7245

# Other contribution

- Correct  xplique/plots/tabular.py to avoid warnings b193084

### CODE REVIEW

File: docs/api/deletion_tab.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Deletion
```
Comment (lucashervier) [inline docs/api/deletion_tab.md:1]: Can you change it to DeletionTab please ? :)

File: tests/utils.py [hunk 2]
```diff
@@ -42,8 +42,8 @@ def generate_regression_model(features_shape, output_shape=1):
     model.add(Dense(4, activation='relu'))
     model.add(Dense(4, activation='relu'))
     model.add(Dense(output_shape))
-    model.compile(loss='mean_absolute_error',
-                  optimizer='sgd')
+    model.compile(loss='mean_absolute_error', optimizer='sgd',
+                  metrics=['accuracy'])
```
Comment (lucashervier) [inline tests/utils.py:46]: I don't believe that having metrics for test is useful. What do you think ? :)

File: xplique/metrics/fidelity.py [hunk 3]
```diff
@@ -625,3 +625,246 @@ def __init__(self,
                  ):  # pylint: disable=R0913
         super().__init__(model, inputs, targets, metric, batch_size,
                          "insertion", baseline_mode, steps, max_percentage_perturbed)
+
+
+class CausalFidelityTab(ExplanationMetric):
```
Comment (lucashervier) [inline xplique/metrics/fidelity.py:630]: I have the feeling that CausalFidelityTab is just a specific CausalFidelityTS. Do you think it is possible to make it inherits from CausalFidelityTS by setting some of the arguments ? It would save a great deal of lines of codes!

File: xplique/plots/tabular.py [hunk 4]
```diff
@@ -199,6 +199,7 @@ def plot_feature_impact(
 
     # add some text for labels and custom y-axis tick labels
     axes.set_xlabel('Impact on output')
+    axes.set_ylabel('')
```
Comment (lucashervier) [inline xplique/plots/tabular.py:202]: Nice catch

**Review by lucashervier — CHANGES_REQUESTED** [review]
I have the intuition that the "Tab" metrics could simply inherits from the "TS" metrics. Therefore, it would save a great deal of files. Do you think its doable ?

File: tests/utils.py (hunk 2 above)
Comment (DavidPetiteau) [reply tests/utils.py:46]: You're right

File: xplique/metrics/fidelity.py [hunk 5]
```diff
@@ -619,9 +606,97 @@ def __init__(self,
...
                  steps: int = 10,
                  max_percentage_perturbed: float = 1.0,
                  ):  # pylint: disable=R0913
         super().__init__(model, inputs, targets, metric, batch_size,
                          "insertion", baseline_mode, steps, max_percentage_perturbed)
+
+
+class DeletionTab(CausalFidelityTS):
+    """
+    Adaptation of the insertion metric for tabular data.
```
Comment (fel-thomas) [inline xplique/metrics/fidelity.py:619]: Insertion or Deletion ? Typo ?

File: xplique/metrics/fidelity.py [hunk 6]
```diff
@@ -401,7 +401,7 @@ def __init__(self,
                  metric: str = "loss",
                  batch_size: Optional[int] = 64,
                  causal_mode: str = "deletion",
-                 baseline_mode: Union[float, str] = 0.0,
+                 baseline_mode: Union[float, Callable] = 0.0,
```
Comment (fel-thomas) [inline xplique/metrics/fidelity.py:404]: So we remove all the baselines mode @AntoninPoche have developed ? Like the negative / inverse / zero ?

File: xplique/plots/tabular.py [hunk 7]
```diff
@@ -199,6 +199,7 @@ def plot_feature_impact(
 
     # add some text for labels and custom y-axis tick labels
     axes.set_xlabel('Impact on output')
```
Comment (fel-thomas) [inline xplique/plots/tabular.py:201]: Is `X` really the impact on output ? I thought it was `X` percent (or 1 - percent) of data that are set to baseline state and `Y` for output score ?

**Review by fel-thomas — CHANGES_REQUESTED** [review]
Great first PR for david! some small changes / things to check that I put in comment. Also you should fix your commits and let's go for merge ;)

File: tests/utils.py (hunk 2 above)
Comment (DavidPetiteau) [reply tests/utils.py:46]: Having metrics for test is useful for causal fidelity for tabular data, otherwise you'll have an error when you test another metric than _loss_ (e.g. _accuracy_):
>        assert metric == "loss" or metric in self.model.metrics_names

File: docs/api/insertion_tab.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
+# Insertion
```
[the remark this answers is not in the document]
Comment (DavidPetiteau) [reply docs/api/insertion_tab.md]: done

File: docs/api/deletion_tab.md (hunk 1 above)
Comment (DavidPetiteau) [reply docs/api/deletion_tab.md]: done

File: xplique/metrics/fidelity.py (hunk 3 above)
Comment (DavidPetiteau) [reply xplique/metrics/fidelity.py]: CausalFidelityTab exists anymore, InsertionTab and DeletionTab inherits from the class CausalFidelityTS

File: xplique/metrics/fidelity.py (hunk 5 above)
Comment (DavidPetiteau) [reply xplique/metrics/fidelity.py]: oups, I change it ... the class is deletion type

File: xplique/metrics/fidelity.py (hunk 6 above)
Comment (DavidPetiteau) [reply xplique/metrics/fidelity.py:404]: I spoke to Antonin, the best way is to have a Callable, one can develop its baseline_mode method
The code is changed in function of the type of the baseline_mode

**Comment by DavidPetiteau** [comment]
that's all for the commits.

**Comment by fel-thomas** [comment]
LGTM !

**Review by fel-thomas — APPROVED** [review]

**Review by lucashervier — APPROVED** [review]
